reactor-core/src/main/java/reactor/core/publisher/Flux.java

```diff
@@ -8955,6 +8969,17 @@ public final void subscribe(Subscriber<? super T> actual) {
 	 * @see #as as() for a loose conversion to an arbitrary type
 	 */
 	public final <V> Flux<V> transformDeferred(Function<? super Flux<T>, ? extends Publisher<V>> transformer) {
+		if (Hooks.DETECT_CONTEXT_LOSS) {
+			String key = CONTEXT_MARKER_PREFIX + UUID.randomUUID().toString();
+			Flux<T> source = this.subscriberContext(ctx -> {
```

Comment:
ideally there wouldn't be two additional `Subscriber` involved here (`subscriberContext`) => merge the `defer` operator and `subscriberContext` subscribers into a dedicated implementation for this particular case?